pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -145,14 +148,35 @@ func init() {
 
 // Add creates a new ClusterDeployment controller and adds it to the manager with default RBAC.
 func Add(mgr manager.Manager) error {
-	return AddToManager(mgr, NewReconciler(mgr))
+	logger := log.WithField("controller", ControllerName)
+	hiveControllersConfig := &corev1.ConfigMap{}
+	if err := mgr.GetAPIReader().Get(context.TODO(), types.NamespacedName{Namespace: controllerutils.GetHiveNamespace(), Name: hive.HiveControllersConfigMapName}, hiveControllersConfig); err != nil {
```

Comment:
Because this repeats in each controller, I think it might still be good to push more down into the controllerutils function. Something like controllerutils.GetControllerConfig taking an apireder and the controller name, and returning all fields and an error. Would reduce a good 20 lines of boilerplate in each.